website/docs/api/pipeline-functions.md

```diff
@@ -130,3 +130,25 @@ exceed the transformer model max length.
 | `min_length`   | The minimum length for a token to be split. Defaults to `25`. ~~int~~ |
 | `split_length` | The length of the split tokens. Defaults to `5`. ~~int~~              |
 | **RETURNS**    | The modified `Doc` with the split tokens. ~~Doc~~                     |
+
+## doc_cleaner {#doc_cleaner tag="function" new="3.2.1"}
+
+Clean up `Doc` attributes. Intended for use at the end of pipelines with
+listeners (`tok2vec`, `transformer`) that store tensors and other values that
+can require a lot of memory and frequently aren't needed after the whole
+pipeline has run.
```

Comment:
It's not the listeners setting these attributes though? Like, a transformer-based component with inline transformer would run into the same mem usage and would also benefit from this cleaner component, no?

Reply:
I'm pretty sure it's actually just the separate pipeline components that set these. They don't really need to be saved otherwise, at least for the pipeline to work. I'll try it out to double-check, though...

I guess this would be a separate issue with an inline transformer not providing the info later like you might expect. And actually, looking at the `transformer` component again, the `DOC_EXT_ATTR` is ignored for the actual setting part, that's not great.

Reply:
I checked, it's really only from the full pipeline components.

Reply:
So for `en_core_web_sm` the `doc.tensor` is from `tok2vec` and not related to the internal `ner` `tok2vec` that ran second.

Reply:
Ah, you're right, ofcourse, it's the `tok2vec` and `transformer` components causing the noise. I'd rephrase this explanation, because this happens regardless of the fact whether there are listeners in the pipeline. While probably unuseful, somebody could have a `transformer` component in the pipeline without a corresponding listener. So maybe something like 

```suggestion
Clean up `Doc` attributes. Intended for use at the end of pipelines with
`tok2vec`/`transformer` pipeline components that store tensors and other values that
can require a lot of memory and frequently aren't needed after the whole
pipeline has run.
```
(+ formatting)

Reply:
Sure, that's fine.

Also unrelated, but it's also really hard to make a pipeline with just a transformers because there are some automatic cleanup steps that remove unused listened-to things. (I find it frustrating because you can't play with just a transformer from `init config`.)